ios/storage.md

```diff
@@ -161,317 +161,620 @@ func downloadData() {
+            }
+            return nil;
+        }
 ```
 
-The AWS AppSync SDK doesn't take a direct dependency on the AWS SDK for iOS for Amazon S3, but takes in `AWSS3TransferUtility` and `AWSS3PresignedURLClient` clients as part of AWSAppSyncClientConfiguration. The code generator used above for generating the API generates the Amazon S3 wrappers required to use the previous clients in the client code. To generate the wrappers, pass the `--add-s3-wrapper` flag while running the code generator tool. You also need to take a dependency on the AWSS3 SDK. You can do that by updating your Podfile to the following:
+### Uploading Binary Data to a File
+
+To upload a binary data to a file, you have to make sure to set the appropriate content type in the uploadData method of the TransferUtility. In the example below, we are uploading a PNG image to S3.
 
-```ruby
-target 'PostsApp' do
-    use_frameworks!
-    pod 'AWSAppSync' ~> '2.6.18'
-    pod 'AWSS3' ~> '2.6.33'
-end
+```swift
+
+let data: Data = Data() // The data to upload
+
+let transferUtility = AWSS3TransferUtility.default()
+transferUtility.uploadData(data,
+          bucket: S3BucketName,
+          key: S3UploadKeyName,
+          contentType: "image/png",
+          expression: expression,
+          completionHandler: completionHandler).continueWith { (task) -> AnyObject! in
+              if let error = task.error {
+                  print("Error: \(error.localizedDescription)")
+              }
+
+              if let _ = task.result {
+                  // Do something with uploadTask.
+              }
+
+              return nil;
+          }
 ```
 
-Then run `pod install` to fetch the new dependency.
-
-Download the updated schema.json from the and put it in the GraphQLOperations folder in the root of the app.
-
-Next, you have to add the new mutation, which is used to perform S3 uploads as part of mutation. Add the following mutation operation in your posts.graphql file:
-
-```graphql
-mutation AddPostWithFile($input: CreatePostWithFileInput!) {
-    createPostWithFile(input: $input) {
-        id
-        title
-        author
-        url
-        content
-        ups
-        downs
-        version
-        file {
-            ...S3Object
+### Downloading Binary Data to a File
+
+The following code shows how to download a binary file.
+
+```swift
+
+let fileURL = // The file URL of the download destination
+let transferUtility = AWSS3TransferUtility.default()
+transferUtility.downloadData(
+        fromBucket: S3BucketName,
+        key: S3DownloadKeyName,
+        expression: expression,
+        completionHandler: completionHandler).continueWith {
+            (task) -> AnyObject! in if let error = task.error {
+                print("Error: \(error.localizedDescription)")
+            }
+
+            if let _ = task.result {
+                // Do something with downloadTask.
+            }
+
+            return nil;
         }
+```
+
+### Limitations
```

Comment:
Let's chat about this section and the next one Monday. I'm not sure I would call this a "Limitations" section as much as a "How it Works" that perhaps we should move higher in this doc. Also I don't know if "Building a Pre-Signed URL" is really needed, nor if we need the SSE section if we already have it in the earlier section on Metadata.

Reply:
Agreed - In fact I think we should remove the sections titled "Downloading to a file", "Uploading binary data to a file", "Downloading binary data to a file"

Reply:
Changed the "Limitations" section to "How it works". Removed  SSE